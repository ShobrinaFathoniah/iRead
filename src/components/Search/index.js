import {StyleSheet, TouchableOpacity, View} from 'react-native';
import React, {useState} from 'react';
import Input from '../Input';
import {moderateScale} from 'react-native-size-matters';
import {LIGHT_BLUE_100, LIGHT_BLUE_100_trans} from '../../helpers/colors';
import {widthPercentageToDP} from 'react-native-responsive-screen';
import Ionicons from 'react-native-vector-icons/Ionicons';

const Search = ({navigation}) => {
  const [search, setSearch] = useState('');

  const goToSearchPage = () => {
    navigation.navigate('Search', {
      params: {search},
    });
  };
  return (
    <View style={styles.searchBar}>
      <Input
        style={styles.input}
        onChangeText={value => setSearch(value)}
        value={search}
      />
      <TouchableOpacity onPress={goToSearchPage}>
        <Ionicons
          style={styles.searchIcon}
          name="search"
          size={40}
          color={LIGHT_BLUE_100}
        />
      </TouchableOpacity>
    </View>
  );
};

export default Search;

const styles = StyleSheet.create({
  searchBar: {
    marginHorizontal: moderateScale(10),
    borderRadius: moderateScale(3),
    marginTop: moderateScale(15),
    flexDirection: 'row',
  },
  input: {
    backgroundColor: LIGHT_BLUE_100_trans,
    borderBottomWidth: moderateScale(0),
    width: widthPercentageToDP(85),
  },
  searchIcon: {
    alignItems: 'center',
    marginTop: moderateScale(3),
  },
});
