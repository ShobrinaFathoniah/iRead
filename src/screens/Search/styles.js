import {StyleSheet, Text, View} from 'react-native';
import {widthPercentageToDP} from 'react-native-responsive-screen';
import {moderateScale} from 'react-native-size-matters';
import {LIGHT_BLUE_100_trans, RED_500} from '../../helpers/colors';

const styles = StyleSheet.create({
  page: {
    flex: 1,
    marginBottom: moderateScale(150),
  },
  searchBar: {
    marginHorizontal: moderateScale(10),
    borderRadius: moderateScale(3),
    marginTop: moderateScale(15),
    flexDirection: 'row',
    alignSelf: 'center',
  },
  input: {
    backgroundColor: LIGHT_BLUE_100_trans,
    borderBottomWidth: moderateScale(0),
    width: widthPercentageToDP(70),
    paddingHorizontal: moderateScale(20),
  },
  searchIcon: {
    alignItems: 'center',
    marginTop: moderateScale(7),
    marginStart: moderateScale(4),
  },
  noDataSearch: {
    alignSelf: 'center',
    color: RED_500,
    fontSize: moderateScale(18),
  },
});

export default styles;
