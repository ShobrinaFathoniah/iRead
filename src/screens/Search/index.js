import {View, TouchableOpacity} from 'react-native';
import React, {useState} from 'react';
import {
  Header,
  Input,
  LibreBaskerville,
  LoadingBar,
  NoConnection,
  Popular,
} from '../../components';
import {navigate} from '../../helpers/navigate';
import styles from './styles';
import Ionicons from 'react-native-vector-icons/Ionicons';
import {LIGTH_BLUE_500} from '../../helpers/colors';
import {useDispatch, useSelector} from 'react-redux';
import axios from 'axios';
import {BASE_URL} from '@env';
import {setIsLoading} from '../../store/globalAction';

const Search = () => {
  const [search, setSearch] = useState('');
  const [dataSearch, setDataSearch] = useState([]);

  const goToHome = () => navigate('Home');
  const dispatch = useDispatch();
  const {dataToken} = useSelector(state => state.login);
  const {isLoading, connection} = useSelector(state => state.global);

  console.log(search);

  const getDataSearch = async () => {
    try {
      dispatch(setIsLoading(true));
      const res = await axios.get(`${BASE_URL}/books?title=${search}`, {
        headers: {Authorization: `Bearer ${dataToken}`},
      });

      console.log(res, 'res');
      console.log(res.data, 'res');
      console.log(`${BASE_URL}/books?title=${search}`);

      setDataSearch(res.data.results);

      dispatch(setIsLoading(false));
    } catch (error) {
      console.log(error);
      dispatch(setIsLoading(false));
    }
  };

  const searchDataScreen = () => {
    return <Popular hideTitle={true} data1={dataSearch} />;
  };

  return (
    <View style={styles.page}>
      <Header button={true} nameIcon="home" onPressButton={goToHome} />
      <View style={styles.searchBar}>
        <Input
          style={styles.input}
          onChangeText={value => {
            setSearch(value);
          }}
          value={search}
        />
        <TouchableOpacity onPress={() => getDataSearch(search)}>
          <Ionicons
            style={styles.searchIcon}
            name="search"
            size={30}
            color={LIGTH_BLUE_500}
          />
        </TouchableOpacity>
      </View>
      {LoadingBar(isLoading)}

      {dataSearch.length === 0 && connection && !isLoading ? (
        <LibreBaskerville style={styles.noDataSearch}>No Data</LibreBaskerville>
      ) : null}
      {connection ? searchDataScreen() : NoConnection(connection)}
    </View>
  );
};

export default Search;
