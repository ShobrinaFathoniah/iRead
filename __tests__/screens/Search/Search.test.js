import * as React from 'react';
import Search from '../../../src/screens/Search';
import {create} from 'react-test-renderer';
import ContainerTesting from '../../../src/helpers/reduxTesting';

test('Search Snapshoot', () => {
  const snap = create(ContainerTesting(<Search />));
  expect(snap).toMatchSnapshot();

  jest.useFakeTimers();
});
