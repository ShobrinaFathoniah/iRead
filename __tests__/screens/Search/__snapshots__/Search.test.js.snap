// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Snapshoot 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "marginBottom": 172.85714285714283,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "marginStart": 23.57142857142857,
          "marginVertical": 23.57142857142857,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <Text
          ellipsizeMode="tail"
          numberOfLines={0}
          style={
            Array [
              Object {
                "fontFamily": "LibreBaskerville-Regular",
              },
              Object {
                "color": "#272b5a",
                "fontSize": 36.14285714285714,
                "marginBottom": 7.857142857142857,
                "marginEnd": 7.857142857142857,
              },
            ]
          }
        >
          Search
        </Text>
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#272b5a",
                "fontSize": 30,
              },
              undefined,
              Object {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
      <Text
        ellipsizeMode="tail"
        numberOfLines={0}
        style={
          Array [
            Object {
              "fontFamily": "LibreBaskerville-Regular",
            },
            Object {
              "color": "#535487",
              "fontSize": 22,
              "marginStart": 7.857142857142857,
              "marginTop": 7.857142857142857,
            },
          ]
        }
      >
        Search your Favorite Book in Here!
      </Text>
    </View>
    <View
      style={
        Object {
          "alignSelf": "center",
          "borderRadius": 4.714285714285714,
          "flexDirection": "row",
          "marginHorizontal": 15.714285714285714,
          "marginTop": 7.857142857142857,
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": "#272b5a",
              "fontSize": 30,
            },
            Object {
              "alignItems": "center",
              "marginRight": -55,
              "marginTop": 11,
            },
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <View
        style={
          Object {
            "alignSelf": "center",
          }
        }
      >
        <TextInput
          onChangeText={[Function]}
          onSubmitEditing={[Function]}
          placeholder="Write in Here"
          placeholderTextColor="#272b5a"
          secureTextEntry={false}
          style={
            Array [
              Object {
                "borderBottomWidth": 1.5714285714285714,
                "borderColor": "#535487",
                "borderRadius": 7.857142857142857,
                "color": "#272b5a",
                "height": 72.28571428571428,
                "marginBottom": 18.857142857142858,
                "padding": 15.714285714285714,
                "width": 424.2857142857143,
              },
              Object {
                "backgroundColor": "#8180b770",
                "borderBottomWidth": 0,
                "paddingStart": 62.857142857142854,
                "width": 525,
              },
            ]
          }
          value=""
        />
      </View>
    </View>
    <View
      style={
        Object {
          "marginVertical": 3.142857142857143,
        }
      }
    >
      <RCTScrollView
        horizontal={true}
        showsHorizontalScrollIndicator={false}
        style={
          Object {
            "marginBottom": 15.714285714285714,
            "marginHorizontal": 15.714285714285714,
          }
        }
      >
        <View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#53548799",
                "borderRadius": 7.857142857142857,
                "height": 62.857142857142854,
                "marginHorizontal": 7.857142857142857,
                "opacity": 1,
                "padding": 15.714285714285714,
              }
            }
            testID="button"
          >
            <Text
              ellipsizeMode="tail"
              numberOfLines={0}
              style={
                Array [
                  Object {
                    "fontFamily": "LibreBaskerville-Regular",
                  },
                  Object {
                    "color": "#000000",
                  },
                ]
              }
            >
              Harry Potter
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#53548799",
                "borderRadius": 7.857142857142857,
                "height": 62.857142857142854,
                "marginHorizontal": 7.857142857142857,
                "opacity": 1,
                "padding": 15.714285714285714,
              }
            }
            testID="button"
          >
            <Text
              ellipsizeMode="tail"
              numberOfLines={0}
              style={
                Array [
                  Object {
                    "fontFamily": "LibreBaskerville-Regular",
                  },
                  Object {
                    "color": "#000000",
                  },
                ]
              }
            >
              Anna Karenina
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#53548799",
                "borderRadius": 7.857142857142857,
                "height": 62.857142857142854,
                "marginHorizontal": 7.857142857142857,
                "opacity": 1,
                "padding": 15.714285714285714,
              }
            }
            testID="button"
          >
            <Text
              ellipsizeMode="tail"
              numberOfLines={0}
              style={
                Array [
                  Object {
                    "fontFamily": "LibreBaskerville-Regular",
                  },
                  Object {
                    "color": "#000000",
                  },
                ]
              }
            >
              Comedy
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#53548799",
                "borderRadius": 7.857142857142857,
                "height": 62.857142857142854,
                "marginHorizontal": 7.857142857142857,
                "opacity": 1,
                "padding": 15.714285714285714,
              }
            }
            testID="button"
          >
            <Text
              ellipsizeMode="tail"
              numberOfLines={0}
              style={
                Array [
                  Object {
                    "fontFamily": "LibreBaskerville-Regular",
                  },
                  Object {
                    "color": "#000000",
                  },
                ]
              }
            >
              Paper
            </Text>
          </View>
        </View>
      </RCTScrollView>
    </View>
    <View
      style={
        Object {
          "marginBottom": 15.714285714285714,
        }
      }
    >
      <RCTScrollView
        ListHeaderComponent={
          <View>
            <RecommendedBook
              hideTitle={true}
            />
          </View>
        }
        data={Array []}
        getItem={[Function]}
        getItemCount={[Function]}
        horizontal={false}
        keyExtractor={[Function]}
        onContentSizeChange={[Function]}
        onLayout={[Function]}
        onMomentumScrollBegin={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        refreshControl={<RefreshControlMock />}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={50}
        showsVerticalScrollIndicator={false}
        stickyHeaderIndices={Array []}
        viewabilityConfigCallbackPairs={Array []}
      >
        <RCTRefreshControl />
        <View>
          <View
            onLayout={[Function]}
          >
            <View>
              <View>
                <RCTScrollView
                  getItem={[Function]}
                  getItemCount={[Function]}
                  horizontal={true}
                  keyExtractor={[Function]}
                  onContentSizeChange={[Function]}
                  onLayout={[Function]}
                  onMomentumScrollBegin={[Function]}
                  onMomentumScrollEnd={[Function]}
                  onScroll={[Function]}
                  onScrollBeginDrag={[Function]}
                  onScrollEndDrag={[Function]}
                  removeClippedSubviews={false}
                  renderItem={[Function]}
                  scrollEventThrottle={50}
                  showsHorizontalScrollIndicator={false}
                  stickyHeaderIndices={Array []}
                  viewabilityConfigCallbackPairs={Array []}
                >
                  <View />
                </RCTScrollView>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>
</View>
`;
